### PR TITLE
Allow list of domains to communicate with server

### PR DESCRIPTION
**Pull request by fmorency**
## Description

Adds a command-line argument to give domain(s) allowed to communicate with the server instance. 

- This is only supported for WebAuthn requests.

## Related Issue

Related to https://github.com/the-lifted-initiative/many-rs/issues/68

Depends on https://github.com/the-lifted-initiative/many-rs/pull/69

## Testing

Manual tests for the moment with a modified Albert version, as well as with the `many` CLI.

## Checklist:

- [x] I have read and followed the CONTRIBUTING guidelines for this project.
- [x] I have added or updated tests and they pass.
- [x] I have added or updated documentation and it is accurate.
- [x] I have noted any breaking changes in this module or downstream modules.

### CODE REVIEW

**Comment by fmorency** [comment]
@hansl, I left this PR in a draft, so you don't merge it accidentally. I will need to update this PR when https://github.com/the-lifted-initiative/many-rs/pull/69 is merged.

File: src/many-ledger/src/main.rs [hunk 1]
```diff
@@ -49,6 +49,12 @@ struct Opts {
     /// If this is not specified the initial state will not be used.
     #[clap(long, short)]
     clean: bool,
+
+    /// Application absolute URLs allowed to communicate with this server. Any
+    /// application will be able to communicate with this server if left empty.
+    /// Multiple occurences of this argument can be given.
+    #[clap(long)]
+    allow: Option<Vec<ManyUrl>>,
```
Comment (hansl) [inline src/many-ledger/src/main.rs:57]:
```suggestion
    allow_origin: Option<Vec<ManyUrl>>,
```

Comment (hansl) [inline src/many-ledger/src/main.rs:57]:
```suggestion
    allow_origin: Option<Vec<ManyUrl>>,
```

File: src/many-ledger/src/main.rs [hunk 2]
```diff
@@ -61,6 +67,7 @@ fn main() {
         mut state,
         persistent,
         clean,
+        allow,
```
Comment (hansl) [inline src/many-ledger/src/main.rs:70]:
```suggestion
        allow_origin,
```

Comment (hansl) [inline src/many-ledger/src/main.rs:70]:
```suggestion
        allow_origin,
```

File: src/many-ledger/src/main.rs [hunk 3]
```diff
@@ -106,6 +113,7 @@ fn main() {
         "many-ledger",
         key,
         Some(std::env!("CARGO_PKG_VERSION").to_string()),
+        allow,
```
Comment (hansl) [inline src/many-ledger/src/main.rs:116]:
```suggestion
        allow_origin,
```

Comment (hansl) [inline src/many-ledger/src/main.rs:116]:
```suggestion
        allow_origin,
```

**Review by hansl — APPROVED** [review]
Approved but make sure to update cargo.toml to latest.

**Pull request merged**